x-pack/plugins/security_solution/public/common/components/event_details/summary_view.tsx

```diff
@@ -112,83 +122,124 @@ const getDescription = ({
-  timelineId,
+  timelineId: contextId,
   eventId,
+  isDisplayingThreatSummary,
+  isDisplayingThreatInfo,
 }: {
   data: TimelineEventsDetailsItem[];
-  browserFields: BrowserFields;
+  browserFields?: BrowserFields;
   timelineId: string;
   eventId: string;
+  isDisplayingThreatSummary: boolean;
+  isDisplayingThreatInfo: boolean;
 }) => {
-  return data != null
-    ? fields.reduce<Summary>((acc, item) => {
-        const field = data.find((d) => d.field === item.id);
-        if (!field) {
-          return acc;
-        }
-        const linkValueField =
-          item.linkField != null && data.find((d) => d.field === item.linkField);
-        const linkValue = getOr(null, 'originalValue.0', linkValueField);
-        const value = getOr(null, 'originalValue.0', field);
-        const category = field.category;
-        const fieldType = get(`${category}.fields.${field.field}.type`, browserFields) as string;
-        const description = {
-          contextId: timelineId,
-          eventId,
-          fieldName: item.id,
-          value,
-          fieldType: item.fieldType ?? fieldType,
-          linkValue: linkValue ?? undefined,
-        };
-
-        if (item.id === 'signal.threshold_result.terms') {
-          try {
-            const terms = getOr(null, 'originalValue', field);
-            const parsedValue = terms.map((term: string) => JSON.parse(term));
-            const thresholdTerms = (parsedValue ?? []).map(
-              (entry: { field: string; value: string }) => {
-                return {
-                  title: `${entry.field} [threshold]`,
-                  description: {
-                    ...description,
-                    value: entry.value,
-                  },
-                };
-              }
-            );
-            return [...acc, ...thresholdTerms];
-          } catch (err) {
-            return acc;
-          }
-        }
+  if (!data) return [];
+  if (isDisplayingThreatSummary || isDisplayingThreatInfo) {
+    return data.reduce<Summary>((acc, { category, field, originalValue }) => {
+      if (isDisplayingThreatInfo && field === 'threat.indicator' && originalValue) {
+        const threatData = Array.isArray(originalValue)
+          ? originalValue.length === 1
+            ? JSON.parse(originalValue[0])
+            : originalValue.map((threatDataItem) => JSON.parse(threatDataItem))
+          : JSON.parse(originalValue);
+        return getDataFromSourceHits(threatData).map((threatInfoItem) => ({
+          title: threatInfoItem.field,
+          description: {
+            contextId,
+            eventId,
+            fieldName: threatInfoItem.field,
+            value: threatInfoItem.originalValue[0],
+          },
+        }));
+      } else if (
+        isDisplayingThreatSummary &&
+        field.startsWith('threat.indicator.') &&
+        originalValue
+      ) {
+        return [
+          ...acc,
+          {
+            title: field.substring('threat.indicator.'.length),
```

Comment:
I don't know if it's this code specifically, but the filtering and "send to timeline functionality does not work with the Threat Info fields:

### Example
<kbd><img width="386" alt="30058990-90e2-11eb-8c1f-71bc0bba3ed5_-_Kibana" src="https://user-images.githubusercontent.com/657252/112911637-77e96280-90bb-11eb-8235-ab492124f315.png"></kbd>

It appears that, unlike the Threat Summary fields, these field names are being taken literally, so that filtering on `0.provider` will create a filter of `0.provider: "geenenesp"` instead of `threat.indicator.provider: "geenenesp"` as expected. The same is true for sending them to the timeline.

Reply:
Additionally, I think that we should add the true fully-qualified field name as a tooltip/alt text for these fields; without it, the user has to do some mental math to figure out which field they're looking at. Luckily, we've already got examples of this on the Alerts table for e.g. signal.rule.name: <kbd>
<img width="500" alt="Fullscreen_3_29_21__6_23_PM" src="https://user-images.githubusercontent.com/657252/112912038-3d33fa00-90bc-11eb-8c24-7d08bdd065a8.png">
</kbd>

Reply:
so I was able to fix the bugs mentioned above, but there is still a piece of the puzzle missing.

by updating the fieldName value on the description object (line 152 in summary_view.tsx) you can now drag the expected key-value pair into the timeline view. Also, on hover, you can now see the expected key. 

However, timeline does not offer any alerts even though the correct key-value pair is dragged into the area. I believe this might be due to the threat.indicator JSON not being deserialized, and therefore it is not possible to filter on anything that would be within the threat.indicator JSON string. 

I am not entirely sure why this field (threat.indicator) is left out as a string JSON while all the other fields are deserialized as expected. This was also the reason why I brought the `getDataFromFieldsHits` to `common` at the first place - to format the JSON after I parse it myself, as it is not already parsed & formatted like the other fields. I briefly considered the implications of parsing threat.indicator JSON earlier (in the backend) but I thought that would be a most likely breaking change, especially considering fields like `first_seen` can appear both as a part of the threat.indicator JSON as well as an actual field on the threat.indicator object (which are two separate entities on type TimelineEventsDetailsItem data (line 125 in summary_view.tsx)). It looks like the parsed & formatted first_seen (that appears on Threat Summary) relates to when the user first sees the threat in their system, while anything within the JSON (appears on Threat Details) is related to filebeat data regarding when the threat was first_seen in the world - and I obviously would not want to override anything. 

All of this being said, I would appreciate some architectural guidance from someone who is specifically familiar with these fields. For now, I am leaving the fieldName fix, but I don't yet have the fix for timeline alert filter not functioning.

Reply:
@ecezalp I believe the crux of your issue is that these are nested fields and so a "normal" query will not show results. I believe that there is logic in the Table view to recognize nested fields and generate nested queries for those fields; I think we need to use that or similar logic for the fields in this panel, since they should all be nested fields under `threat.indicator`.

Reply:
hmm interestingly enough, from the table view, it is not possible to drag the threat.indicator JSON (or any part of it) into the timeline view at the first place. this makes me think that timeline does not support what I attempted to do (filtering alerts on the key-value pairs within the threat.indicator JSON). If that's the case, we should probably build that functionality in a separate ticket, and it could be a great win